src/common/src/types/interval.rs

```diff
@@ -65,22 +66,160 @@ impl Interval {
+    /// Returns the number of microseconds.
+    ///
+    /// Note the difference between [`usecs`] and [`seconds_in_micros`].
+    /// We have:
+    /// `usecs` = `hours_field` * 3600
+    ///     + `minutes_field` * 60
+    ///     + `seconds_in_micros` as f64 / 1_000_000.0
+    pub fn usecs(&self) -> i64 {
         self.usecs
     }
 
-    pub fn get_usecs_of_day(&self) -> u64 {
+    /// Calculates the remaining number of microseconds.
+    /// range: 0..86_400_000_000
+    ///
+    /// Note the difference between [`usecs`] and [`usecs_of_day`].
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "-1:00:00".parse().unwrap();
+    /// assert_eq!(interval.usecs(), -1 * 60 * 60 * 1_000_000);
+    /// assert_eq!(interval.usecs_of_day(), 23 * 60 * 60 * 1_000_000);
+    /// ```
+    pub fn usecs_of_day(&self) -> u64 {
         self.usecs.rem_euclid(USECS_PER_DAY) as u64
     }
 
+    /// Returns the years field. range: unlimited
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "2332 yrs 12 months".parse().unwrap();
+    /// assert_eq!(interval.years_field(), 2333);
+    /// ```
+    pub fn years_field(&self) -> i32 {
+        self.months / 12
+    }
+
+    /// Returns the months field. range: -11..=11
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "15 months".parse().unwrap();
+    /// assert_eq!(interval.months_field(), 3);
+    ///
+    /// let interval: Interval = "-15 months".parse().unwrap();
+    /// assert_eq!(interval.months_field(), -3);
+    /// ```
+    pub fn months_field(&self) -> i32 {
+        self.months % 12
+    }
+
+    /// Returns the days field. range: unlimited
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "1 months 100 days 25:00:00".parse().unwrap();
+    /// assert_eq!(interval.days_field(), 100);
+    /// ```
+    pub fn days_field(&self) -> i32 {
+        self.days
+    }
+
+    /// Returns the hours field. range: unlimited
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "25:00:00".parse().unwrap();
+    /// assert_eq!(interval.hours_field(), 25);
+    ///
+    /// let interval: Interval = "-25:00:00".parse().unwrap();
+    /// assert_eq!(interval.hours_field(), -25);
+    /// ```
+    pub fn hours_field(&self) -> i32 {
+        (self.usecs / USECS_PER_SEC / 3600) as i32
+    }
+
+    /// Returns the minutes field. range: -59..=-59
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "00:20:00".parse().unwrap();
+    /// assert_eq!(interval.minutes_field(), 20);
+    ///
+    /// let interval: Interval = "-00:20:00".parse().unwrap();
+    /// assert_eq!(interval.minutes_field(), -20);
```

Comment:
It's in check step

Reply:
Sorry for my mistake. 😇 Should we move it to the test stage?

Reply:
Another parsing bug ... The negative sign got lost because hour field is zero ...
https://github.com/risingwavelabs/risingwave/blob/bccc76de718d55808e5c88205223f848eba31aea/src/common/src/types/interval.rs#L1088-L1093

Reply:
fixed in bd1a21a